extensions/adobe/experience/campaign/orchestration/reportingevent.schema.json

```diff
@@ -0,0 +1,158 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id":
+    "https://ns.adobe.com/experience/campaign/orchestration/reportingevent",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "User journey reporting event",
+  "type": "object",
+  "description": "",
+  "definitions": {
+    "reportingevent": {
+      "properties": {
+        "xdm:timestamp": {
+          "title": "Timestamp for reporting event",
+          "type": "string",
+          "format": "date-time",
+          "description": "Time of generation of reporting event"
+        },
+        "xdm:dataEnrichmentSource": {
+          "title": "Data Source",
+          "$ref": "https://ns.adobe.com/xdm/data/datasource",
+          "description": "Data source used for data enrichment"
+        },
+        "xdm:dataEntityUID": {
+          "title": "Unique identifier of data entity",
+          "type": "string",
+          "format": "uri",
+          "description":
+            "The identifier for the data entity used for data enrichment"
+        },
+        "xdm:orgID": {
+          "title": "Organizational unit identifier",
+          "type": "string",
+          "description":
+            "Identifier of the organizational unit for this journey version"
+        },
+        "xdm:flowControlType": {
+          "title": "Flow control",
+          "type": "string",
+          "description": "Flow control in the journey triggered by the event.",
+          "meta:enum": {
```

Comment:
Is this a hard `enum` or a soft enum? If the values defined in `meta:enum` are the only permissible values, then it's a hard enum, and you should add an `enum` array.

In addition, in XDM, it is best practice to use lowercase snake_case names for enum values (or URIs), this might be something for you to follow.

Making it an `enum` does not affect compatibility, unless you plan to remove `enum` values later on.

Reply:
The enum values are for now only we permit. It is likely to have additions in it but removal is something I can't foresee. I will discuss on this with team and make the changes for it
Also made changes to use snake_case for enum in all places

Reply:
Made as hard enum now